### PR TITLE
Jetpack Manage: Adding a new Manage Pricing page

### PR DESCRIPTION
**Pull request by coder-karen**
Related to #https://github.com/Automattic/jetpack-manage/issues/244

## Proposed Changes

* This PR adds a new Jetpack Manage pricing page at `/manage/pricing`.
* Once this PR is merged the page will only be available in staging and development.
* The page includes a tracking event on page view
* The page also includes the Jetpack.com Masterbar, but the main menu is not sticky unlike on other Jetpack.com pages (because the license selector will be sticky, a later addition).
* Currently the page also includes a header with a sub-heading, a footer, recommendations block and FAQ section, as with the current `cloud.jetpack.com/pricing` page.
* There is no other content at this stage.
* This PR also includes a change to one of the paths defined for the `cloud.jetpack.com/pricing` page in `client/sections.js`, to make sure content from the pricing page doesn't appear on the `/manage/pricing` page.

## Testing Instructions

* To test, check out this PR and visit `http://jetpack.cloud.localhost:3000/manage/pricing`. What you see should be similar to the below screenshot.
* You should also see a `calypso_jetpack_manage_pricing_visit` firing.

Screenshot 2024-01-23 at 16 58 35

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by gogdzl — APPROVED** [review]
It works!

File: client/jetpack-cloud/sections/manage/pricing/primary/index.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+				wideLayout
+			>
+				<DocumentHead
+					title={ translate(
+						'Jetpack Manage: Bulk discounts and flexible billing to suit your needs'
+					) }
+				/>
+				<div className="jetpack-product-store">
+					<Header />
+					<Recommendations />;
```
Comment (coder-karen) [inline client/jetpack-cloud/sections/manage/pricing/primary/index.tsx]:
```suggestion
					<Recommendations />
```
Unnecessary ';', needs removing.

File: config/jetpack-cloud-horizon.json [hunk 2]
```diff
@@ -81,6 +81,7 @@
 		"jetpack-cloud-partner-portal": true,
 		"jetpack-cloud-plugin-management": true,
 		"jetpack-cloud-pricing": true,
+		"jetpack-cloud-manage-pricing": false,
```
Comment (cleacos) [inline config/jetpack-cloud-horizon.json]: If we set this feature flag to true in Horizon, we could test it here in this PR using the Jetpack Cloud live link (like staging).

Comment (cleacos) [reply config/jetpack-cloud-horizon.json]: Here: 
image

Comment (coder-karen) [reply config/jetpack-cloud-horizon.json]: Oh, interesting! I'll try that.

Comment (coder-karen) [inline config/jetpack-cloud-horizon.json]:
```suggestion
		"jetpack-cloud-manage-pricing": true,
```
Setting to true based on a PR review suggestion.

**Review by cleacos — COMMENTED** [review]
I tested it:

- The design looks good
- The track events are fired for:
   - The `calypso_jetpack_manage_pricing_visit`
   - Open any element on the FAQ list: `calypso_plans_faq_open`
- **Issue:** If I open the Products panel (first option at the top menu), it shows the menu, but it's not sticky to the window, if I scroll down the products panel is moved together the content.

File: config/jetpack-cloud-horizon.json (hunk 2 above)
Comment (cleacos) [reply config/jetpack-cloud-horizon.json]: Thanks Karen!

**Comment by coder-karen** [comment]
Thanks for testing @cleacos !

> Issue: If I open the Products panel (first option at the top menu), it shows the menu, but it's not sticky to the window, if I scroll down the products panel is moved together the content.

That's a good point. Because the masterbar isn't sticky anymore, the behaviour for that link is different. @keoshi what do you think about this - should we somehow modify the masterbar styles just for `manage/pricing#products`? I'm not sure whether / how that would be possible initially but can look into it if it is an issue.

**Review by keoshi — CHANGES_REQUESTED** [review]
Looks good to me! Noticed a couple of differences from the main pricing page in production:

1. There's a unnecessary padding at the top due to the styles in `.theme-jetpack-cloud .layout.has-no-masterbar`.
2. Because the top nav isn't sticky, the spacing in `.is-group-jetpack-cloud.is-section-jetpack-cloud-manage-pricing .header` pushes the content down. Setting the `top-margin` to 0 seems to fix that.

**Comment by keoshi** [comment]
> @keoshi what do you think about this - should we somehow modify the masterbar styles just for manage/pricing#products? I'm not sure whether / how that would be possible initially but can look into it if it is an issue.

Uff, didn't even catch this one. Thanks @cleacos for bringing it up.

If possible, when the menu is open I'd make scrolling down the page impossible via `overflow-y: hidden` or something like that. It's sort of hacky and not ideal, but if we want to add a sticky license number selector later on, it's the only option I can think of right now.

**Comment by coder-karen** [comment]
Thanks @keoshi! I've added these to a new follow-on issue so that we can get the page code merged and be able to move forward on other tasks too: https://github.com/Automattic/jetpack-manage/issues/253

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/11009197

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @coder-karen for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/11009197).

**Pull request merged**